### PR TITLE
Timeline a11y

### PR DESCRIPTION
**Pull request by charlag**
(After Tusky 5)
Apply suggestions from #1003.
Does only minimal changes, there needs to be more done for threads and notifications (also no way to refresh timeline now).

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/util/ListStatusAccessibilityDelegate.kt [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+import androidx.core.view.AccessibilityDelegateCompat
+import androidx.core.view.accessibility.AccessibilityNodeInfoCompat
+import androidx.core.view.accessibility.AccessibilityNodeInfoCompat.AccessibilityActionCompat
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerViewAccessibilityDelegate
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.interfaces.StatusActionListener
+import com.keylesspalace.tusky.viewdata.StatusViewData
+
+// Not using lambdas because there's boxing of int then
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/util/ListStatusAccessibilityDelegate.kt]: did not know that, thx

File: app/src/main/java/com/keylesspalace/tusky/adapter/StatusViewHolder.java [hunk 2]
```diff
@@ -92,11 +95,88 @@ protected void setupWithStatus(StatusViewData.Concrete status, final StatusActio
             }
 
             rebloggedBar.setOnClickListener(v -> listener.onOpenReblog(getAdapterPosition()));
+        }
+        setContentDescription(status);
+        // Workaround for RecyclerView 1.0.0 / androidx.core 1.0.0
+        // RecyclerView tries to set AccessibilityDelegateCompat to null
+        // but ViewCompat code replaces is with the default one. RecyclerView never
+        // fetches another one from its delegate because it checks that it's set so we remove it
+        // and let RecyclerView ask for a new delegate.
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/StatusViewHolder.java]: Will this be fixed in a upcoming release of AndroidX?

File: app/src/main/java/com/keylesspalace/tusky/adapter/StatusViewHolder.java [hunk 3]
```diff
@@ -92,11 +95,88 @@ protected void setupWithStatus(StatusViewData.Concrete status, final StatusActio
...
+            itemView.setContentDescription(
+                    itemView.getContext().getString(R.string.load_more_placeholder_text));
+        } else {
+            setStatusDescriptionForStatus(status);
         }
 
     }
 
+    private void setStatusDescriptionForStatus(@NonNull StatusViewData.Concrete status) {
+        CharSequence postedAgo = DateUtils.getRelativeTimeSpanString(
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/StatusViewHolder.java]: Shouldn't this respect the "use absolute time" setting?

File: app/src/main/java/com/keylesspalace/tusky/adapter/StatusViewHolder.java [hunk 4]
```diff
@@ -92,11 +95,88 @@ protected void setupWithStatus(StatusViewData.Concrete status, final StatusActio
...
+                status.getCreatedAt().getTime(),
+                System.currentTimeMillis(),
+                DateUtils.SECOND_IN_MILLIS,
+                DateUtils.FORMAT_ABBREV_RELATIVE
+        );
+        Context context = itemView.getContext();
+
+
+        String description = itemView.getContext().getString(R.string.status_description,
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/StatusViewHolder.java]:
```suggestion
        String description = context.getString(R.string.status_description,
```

File: app/src/main/java/com/keylesspalace/tusky/util/ListStatusAccessibilityDelegate.kt (hunk 1 above)
Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/util/ListStatusAccessibilityDelegate.kt]: In this particular case. Kotlin exports it's lambda interface as generic method to Java. Generics are always boxed (that's why there are `IntStream`  and such).

File: app/src/main/java/com/keylesspalace/tusky/adapter/StatusViewHolder.java (hunk 2 above)
Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/adapter/StatusViewHolder.java]: I didn't find any info anywhere. Didn't report it tbh.

File: app/src/main/java/com/keylesspalace/tusky/adapter/StatusViewHolder.java (hunk 3 above)
Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/adapter/StatusViewHolder.java]: Yep, we probably should.

**Comment by charlag** [comment]
I did not extend it to the detailed status because I was unable to reproduce some native TextView functionality. When you focus TextView, there's an accessibility action to select a specific link. I could reproduce it with mentions but I'm unsure about links, maybe if I could scan for spannables.
Anyway, I could try to reproduce it and then we can decide.

**Comment by charlag** [comment]
I keep description card intentionally, otherwise detailed status behaves like a regular one now.

**Comment by charlag** [comment]
Hey
I've added/fixed some things and it can be reviewed at any point.
I'm not sure if we need to make a build to let people try it out or we can just get feedback from nightly later.

**Comment by connyduck** [comment]
I would merge it after the next bugfix release if thats ok? In the meantime here would be the link for a build from this branch: https://app.bitrise.io/artifact/12488468/p/debe1dde81121cdf1d4514238f094c91

**Comment by charlag** [comment]
@connyduck sure, as you like. That's for the link, I'll share it today!

**Pull request merged**